core/sphinx/kemsphinx_test.go

```diff
@@ -138,13 +165,13 @@ func testForwardKEMSphinx(t *testing.T, mykem kem.Scheme, sphinx *Sphinx, testPa
 
 		// Create the packet.
 		payload := []byte(testPayload)
-		pkt, err := sphinx.NewKEMPacket(rand.Reader, path, payload)
+		pkt, err := sphinx.NewPacket(rand.Reader, path, payload)
 		require.NoError(err, "NewKEMPacket failed")
-		require.Len(pkt, sphinx.Geometry().HeaderLength+sphinx.Geometry().PayloadTagLength+len(payload), "Packet Length")
+		require.Equal(len(pkt), sphinx.Geometry().HeaderLength+sphinx.Geometry().PayloadTagLength+len(payload), "Packet Length")
 
 		// Unwrap the packet, validating the output.
 		for i := range nodes {
-			b, _, cmds, err := sphinx.KEMUnwrap(nodes[i].privateKey, pkt)
+			b, _, cmds, err := sphinx.unwrapKem(nodes[i].privateKey, pkt)
```

Comment:
we're not testing the `replayTag` here. can we do something useful with that? like maybe test that they are different in the various layers or something?

Reply:
Why would we do that? Would we be simulating the behavior of the mix server? Would we be checking to make sure Sphinx is implemented correctly? If we look at how it's implemented we can decide a lot of stuff without writing tests.